operator/builtin/input/aws/cloudwatch/cloudwatch.go

```diff
@@ -0,0 +1,372 @@
+	for {
+		input := c.filterLogEventsInputBuilder(nextToken)
+
+		resp, err := svc.FilterLogEvents(&input)
+		if err != nil {
+			c.Errorf("failed to get events: %s", err)
+			break
+		}
+
+		if len(resp.Events) == 0 {
+			c.Debug("No new events available from AWS Cloudwatch Logs")
+		}
+
+		c.handleEvents(ctx, resp.Events)
+
+		if resp.NextToken == nil {
+			c.Debug("Finished getting events")
+			break
+		}
+		nextToken = *resp.NextToken
+		c.Debugf("Reached event limit '%d'", c.eventLimit)
+	}
+}
+
+// filterLogEventsInputBuilder builds AWS Cloudwatch Logs Filter Log Events Input based on provided values
+// and returns completed input.
+func (c *CloudwatchInput) filterLogEventsInputBuilder(nextToken string) cloudwatchlogs.FilterLogEventsInput {
+	logGroupNamePtr := aws.String(c.logGroupName)
+	limit := aws.Int64(c.eventLimit)
+	startTime := aws.Int64(c.startTime)
+
+	if c.logStreamNamePrefix != "" && nextToken != "" {
+		tmp := c.timeLayoutParser(c.logStreamNamePrefix)
+		logStreamNamePrefixPtr := aws.String(tmp)
+		nextTokenPtr := aws.String(nextToken)
+		return cloudwatchlogs.FilterLogEventsInput{
+			Limit:               limit,
+			LogGroupName:        logGroupNamePtr,
+			LogStreamNamePrefix: logStreamNamePrefixPtr,
+			StartTime:           startTime,
+			NextToken:           nextTokenPtr,
+		}
+	}
+
+	if c.logStreamNamePrefix != "" {
+		tmp := c.timeLayoutParser(c.logStreamNamePrefix)
+		logStreamNamePrefixPtr := aws.String(tmp)
+		return cloudwatchlogs.FilterLogEventsInput{
+			Limit:               limit,
+			LogGroupName:        logGroupNamePtr,
+			LogStreamNamePrefix: logStreamNamePrefixPtr,
+			StartTime:           startTime,
+		}
+	}
+
+	if len(c.logStreamNames) > 0 && nextToken != "" {
+		nextTokenPtr := aws.String(nextToken)
+		return cloudwatchlogs.FilterLogEventsInput{
+			Limit:          limit,
+			LogGroupName:   logGroupNamePtr,
+			LogStreamNames: c.logStreamNames,
+			StartTime:      startTime,
+			NextToken:      nextTokenPtr,
+		}
+	}
+
+	if len(c.logStreamNames) > 0 {
+		return cloudwatchlogs.FilterLogEventsInput{
+			Limit:          limit,
+			LogGroupName:   logGroupNamePtr,
+			LogStreamNames: c.logStreamNames,
+			StartTime:      startTime,
+		}
+	}
+
+	if nextToken != "" {
+		nextTokenPtr := aws.String(nextToken)
+		return cloudwatchlogs.FilterLogEventsInput{
+			Limit:        limit,
+			LogGroupName: logGroupNamePtr,
+			StartTime:    startTime,
+			NextToken:    nextTokenPtr,
+		}
+	}
+
+	return cloudwatchlogs.FilterLogEventsInput{
+		Limit:        limit,
+		LogGroupName: logGroupNamePtr,
+		StartTime:    startTime,
+	}
+}
+
+// handleEvent is the handler for a AWS Cloudwatch Logs Filtered Event.
+func (c *CloudwatchInput) handleEvent(ctx context.Context, event *cloudwatchlogs.FilteredLogEvent) error {
+	e := make(map[string]interface{})
+	e["message"] = event.Message
+	e["event_id"] = event.EventId
+	e["log_stream_name"] = event.LogStreamName
```

Comment:
We spoke about this, but ill leave a comment

I think we should consider promoting event_id, log_stream_name and region as resources.

Reply:
Resource should describe the emitter of the log - so that logs can be grouped by resource. In this case, `log_stream_name` and `region` would seem to do that.

`event_id` would seem to be an attribute, since grouping by this would likely yield just this record.